gravity.sh

```diff
@@ -837,12 +837,76 @@ database_recovery() {
   echo ""
 }
 
+gravity_optimize() {
+    # The ANALYZE command gathers statistics about tables and indices and stores
+    # the collected information in internal tables of the database where the
+    # query optimizer can access the information and use it to help make better
+    # query planning choices
+    local str="Optimizing database"
+    echo -ne "  ${INFO} ${str}..."
+    output=$( { pihole-FTL sqlite3 -ni "${gravityTEMPfile}" "PRAGMA analysis_limit=0; ANALYZE" 2>&1; } 2>&1 )
+    status="$?"
+
+    if [[ "${status}" -ne 0 ]]; then
+        echo -e "\\n  ${CROSS} Unable to optimize database ${gravityTEMPfile}\\n  ${output}"
+        gravity_Cleanup "error"
+    else
+        echo -e "${OVER}  ${TICK} ${str}"
+    fi
+}
+
+# Function: timeit
+# Description: Measures the execution time of a given command.
+#
+# Usage:
+#   timeit <command>
+#
+# Parameters:
+#   <command> - The command to be executed and timed.
+#
+# Returns:
+#   The exit status of the executed command.
+#
+# Output:
+#   If the 'timed' variable is set to true, prints the elapsed time in seconds
+#   with millisecond precision.
+#
+# Example:
+#   timeit ls -l
+#
+timeit(){
+  local start_time end_time elapsed_time ret
+
+  # Capture the start time
+  start_time=$(date +%s%3N)
+
+  # Execute the command passed as arguments
+  "$@"
+  ret=$?
+
+  if [[ "${timed:-}" != true ]]; then
+    return $ret
+  fi
+
+  # Capture the end time
+  end_time=$(date +%s%3N)
+
+  # Calculate the elapsed time
+  elapsed_time=$((end_time - start_time))
+
+  # Display the elapsed time
+  printf "  --> took %d.%03d seconds\n" $((elapsed_time / 1000)) $((elapsed_time % 1000))
```

Comment:
We could add a little bit of color here (blue or maybe gray):
```suggestion
  printf "  %b--> took %d.%03d seconds%b\n" ${COL_BLUE} $((elapsed_time / 1000)) $((elapsed_time % 1000)) ${COL_NC}
```

![image](https://github.com/user-attachments/assets/c44c0939-dc85-4168-befa-1dcac268ad59)